File: Go/1.Two_Sum.go
package main

func twoSum(nums []int, target int) []int {
		result := make(map[int]int)
    for i, num := range nums {
        current := target - num
				if j, exists := result[current]; exists {
					return []int{j, i}
				}

				result[num] = i
    }

		return nil
}

File: Typescript/1.Two_Sum/1.Two_Sum.ts
function twoSum(nums: number[], target: number): number[] | null {
  const result = new Map<number, number>();

  for (const [i, num] of nums.entries()) {
    const current = target - num;

    if (result.has(current)) {
      return [result.get(current)!, i];
    }

    result.set(num, i);
  }

  return null;
}
